common/src/main/java/io/pravega/common/util/btree/BTreeIndex.java

```diff
@@ -693,17 +814,22 @@ private ByteArraySegment generateMinKey() {
         Preconditions.checkArgument(pageCollection.getIndexLength() == offset, "IndexLength mismatch.");
         pages.add(new AbstractMap.SimpleImmutableEntry<>(offset, getFooter(lastPage.getOffset(), lastPage.getPage().getLength())));
 
-        // Also collect the old footer's offset, as it will be replaced by a more recent value.
+        // Collect the old footer's offset, as it will be replaced by a more recent value.
         long oldFooterOffset = getFooterOffset(state.length);
         if (oldFooterOffset >= 0) {
             oldOffsets.add(oldFooterOffset);
         }
 
+        // Collect the offsets of removed pages. These are no longer needed.
+        pageCollection.collectRemovedPageOffsets(oldOffsets);
```

Comment:
I only found this method from the `UpdateablePageCollection` being invoked. I'm trying to understand how the `Updateable` version of `PageCollection` is being used, and I was trying to track the new methods defined there.

Reply:
Sure, I get that. Let me elaborate on what I'm trying to understand. I searched for utilizations of methods like:

```
collectRemovedPageOffsets(Collection<Long> target)
collectLeafPages(Collection<PageWrapper> target)
collectPages(Collection<Long> offsets, Collection<PageWrapper> target)
```

and I couldn't find them, so I was wondering what is going on. Are those methods invoked and I missed it? Are you planning on using them in the future?

Reply:
Yes they are used. It's hard to find them on GitHub, but with the IDE it's much easier. 
- `collectRemovedPageOffsets` - BTreeIndex line 824.
- `collectLeafPages` - BTreeIndex line 449.
- `collectPages` - BtreeIndex line 484.

Reply:
Ok.